code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,157 @@
+#define JOB_CHOICE_YES "Yes"
+#define JOB_CHOICE_REROLL "Reroll"
+#define JOB_CHOICE_CANCEL "Cancel"
+
+GLOBAL_DATUM_INIT(latejoin_menu, /datum/latejoin_menu, new)
+
+/// Makes a list of jobs and pushes them to a DM list selector. Just in case someone did a special kind of fucky-wucky with TGUI.
+/datum/latejoin_menu/proc/fallback_ui(mob/dead/new_player/user)
+	var/list/jobs = list()
+	for(var/datum/job/job in SSjob.joinable_occupations)
+		jobs += job.title
+
+	var/input = tgui_input_list(user, "Pick a job to join as:", "Latejoin Job Selection", jobs)
+
+	if(!input)
+		return
+
+	user.AttemptLateSpawn(input)
+
+/datum/latejoin_menu/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "JobSelection", "Latejoin Menu")
+		ui.open()
+
+/datum/latejoin_menu/ui_data(mob/user)
+	var/mob/dead/new_player/owner = user
+	var/list/departments = list()
+	var/list/data = list(
+		"disable_jobs_for_non_observers" = SSlag_switch.measures[DISABLE_NON_OBSJOBS],
+		"round_duration" = DisplayTimeText(world.time - SSticker.round_start_time),
+		"departments" = departments,
+	)
+	if(SSshuttle.emergency)
+		switch(SSshuttle.emergency.mode)
+			if(SHUTTLE_ESCAPE)
+				data += list("shuttle_status" = "The station has been evacuated.")
+			if(SHUTTLE_CALL)
+				if(!SSshuttle.canRecall())
+					data += list("shuttle_status" = "The station is currently undergoing evacuation procedures.")
+
+	for(var/datum/job/prioritized_job in SSjob.prioritized_jobs)
+		if(prioritized_job.current_positions >= prioritized_job.total_positions)
+			SSjob.prioritized_jobs -= prioritized_job
+
+	for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+		var/list/department_jobs = list()
+		var/list/department_data = list(
+			"name" = department.department_name,
+			"jobs" = department_jobs,
+			"color" = department.ui_color,
+			"open_slots" = 0,
+		)
+		departments += list(department_data)
+
+		for(var/datum/job/job_datum as anything in department.department_jobs)
+			var/job_availability = owner.IsJobUnavailable(job_datum.title, TRUE)
+			var/datum/outfit/outfit = job_datum.outfit
+			var/datum/id_trim/trim = initial(outfit.id_trim)
+
+			var/list/job_data = list(
+				"command" = !!(job_datum.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND),
+				"description" = job_datum.description,
+				"prioritized" = !!(job_datum in SSjob.prioritized_jobs),
```

Comment:
```suggestion
				"prioritized" = job_datum in SSjob.prioritized_jobs,
```

Reply:
Oh, right, I keep forgetting about that, I'm so used to bools being returned for code like that in other languages.